woocommerce/woocommerce-admin
# Update @wordpress/eslint-plugin dependency to latest version.

This pull request updates the `@wordpress/eslint-plugin` dependency for the `@woocommerce/eslint-plugin` to the latest version (`8.0.0`) from `7.1.0`. 

With these changes:

- A new "eslint-plugin-import" dependency on the main package file due to changes in eslint.
- moving `prettier` and `@wordpress/prettier-config` dependencies from main package.json to the `@woocommerce/eslint-plugin` package.json. This reduces the imports needed by consumers of `@woocommerce/eslint-plugin` hopefully.
- a jsdoc lint plugin update considers `typescript` style jsdocs as errors by default. So I added a setting configuration update to the `@woocommerce/eslint-plugin` config to set jsdoc style to `typescript`. 
- A bunch of new linting rules came with the update so I temporarily converted errors to warnings for followups after merge. This allows getting the package update released sooner and followups can make changes carefully.

## To Test

- There should be no linting errors reported as a result of these changes.
- Verify that eslint fix and prettier format fix still work cli and in your IDE (both worked great for me - I use VSCode).
